delta/service/src/test/scala/ch/epfl/bluebrain/nexus/delta/service/projects/ProjectImplSpec.scala

```diff
@@ -0,0 +1,106 @@
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{ProjectEvent, ProjectRef}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, Label}
+import ch.epfl.bluebrain.nexus.delta.sdk.testkit.{AclsDummy, PermissionsDummy, ProjectsBehaviors}
+import ch.epfl.bluebrain.nexus.delta.sdk.{Permissions, Projects}
+import ch.epfl.bluebrain.nexus.delta.service.utils.EventLogUtils
+import ch.epfl.bluebrain.nexus.delta.service.{AbstractDBSpec, ConfigFixtures}
+import ch.epfl.bluebrain.nexus.sourcing.EventLog
+import monix.bio.UIO
+import org.scalatest.OptionValues
+
+class ProjectImplSpec extends AbstractDBSpec with ProjectsBehaviors with OptionValues with ConfigFixtures {
+
+  val serviceAccount: Subject        = Identity.User("serviceAccount", Label.unsafe("realm"))
+  val projectsConfig: ProjectsConfig = ProjectsConfig(aggregate, keyValueStore, pagination, indexing)
+
+  val ownerPermissions = Set(
+    Permissions.projects.read,
+    resources.read,
+    resources.write,
+    resolvers.write,
+    views.write,
+    views.query,
+    schemas.write,
+    files.write,
+    storages.write
+  )
+
+  // A project created on org1 has all owner permissions on / and org1
+  val (rootPermissions, org1Permissions) = ownerPermissions.splitAt(ownerPermissions.size / 2)
+  val proj10                             = Label.unsafe("proj10")
+
+  // A project created on org2 lacks some of the owner permissions
+  // proj20 has all owner permissions on /, org2 and proj20
+  val proj20                               = Label.unsafe("proj20")
+  val (org2Permissions, proj20Permissions) = org1Permissions.splitAt(org1Permissions.size / 2)
+
+  // proj21 has some extra acls that should be conserved
+  val proj21            = Label.unsafe("proj21")
+  val proj21Permissions = Set(orgs.read, orgs.create)
+
+  // proj22 has no permission set
+  val proj22 = Label.unsafe("proj22")
+
+  val acls = {
+    for {
+      acls <- AclsDummy(PermissionsDummy(PermissionsGen.minimum))
+      _    <- acls.append(AclAddress.Root, Acl(subject -> rootPermissions), 0L)(serviceAccount)
+      _    <- acls.append(AclAddress.Organization(org1), Acl(subject -> org1Permissions), 0L)(serviceAccount)
+      _    <- acls.append(AclAddress.Organization(org2), Acl(subject -> org2Permissions), 0L)(serviceAccount)
+      _    <- acls.append(AclAddress.Project(org2, proj20), Acl(subject -> proj20Permissions), 0L)(serviceAccount)
+      _    <- acls.append(AclAddress.Project(org2, proj21), Acl(subject -> proj21Permissions), 0L)(serviceAccount)
+    } yield acls
+  }.accepted
+
+  override def create: UIO[Projects] =
+    for {
+      eventLog <- EventLog.postgresEventLog[Envelope[ProjectEvent]](EventLogUtils.toEnvelope).hideErrors
+      orgs     <- organizations
+      projects <- ProjectsImpl(projectsConfig, eventLog, orgs, acls, ownerPermissions, serviceAccount)
+    } yield projects
+
+  "Creating projects" should {
```

Comment:
I'm wondering if the Projects dummy should have the exact same behaviour... why would we keep the acls application only in the service impl?

Reply:
It seems out of scope of the dummy applying write actions downwards to other dummys.
And this common code would have to go in the sdk which feels a little weird too.